src/DataLevelService.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+namespace Drupal\commerce_bluesnap;
+
+use Drupal\commerce_store\Entity\StoreInterface;
+use Drupal\commerce_order\Entity\OrderInterface;
+use Drupal\commerce_order\Entity\OrderItemInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Bluesnap data level service class.
+ *
+ * Provides level 2 and level 3 transaction data.
+ */
+class DataLevelService {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildDataLevelSettingsForm(StoreInterface $store) {
+    $form = [];
+
+    // Get bluesnap data level settings.
+    $settings = $this->getDataLevelSetting($store);
+
+    $form['bluesnap_data_level_settings'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Bluesnap data level settings'),
+      '#tree' => TRUE,
+      '#open' => TRUE,
+    ];
+    $form['bluesnap_data_level_settings']['status'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable bluesnap level 2/3 data processing'),
+      '#default_value' => $settings ? $settings->status : '0',
+    ];
+    $form['bluesnap_data_level_settings']['type'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Bluesnap data processing level'),
+      '#options' => [
+        'level_2' => $this->t('Level 2'),
+        'level_3' => $this->t('Level 3'),
+      ],
+      '#states' => [
+        'visible' => [
+          ':input[name="bluesnap_data_level_settings[bluensnap_data_level_status]"]' => [
+            'checked' => TRUE,
+          ],
+        ],
+      ],
+      '#default_value' => $settings ? $settings->type : 'level_2',
+    ];
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDataLevelSetting(StoreInterface $store) {
+    $settings = $store->get('bluesnap_data_level_settings')->value;
+    $settings = json_decode($settings);
+    return $settings;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCardSuppprtLevel3() {
+    return ['mastercard', 'visa'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCardSuppprtLevel2() {
```

Comment:
Do these need to be public functions?

Reply:
Those can actually be private. Made them private